pinot-common/src/main/java/org/apache/pinot/common/utils/OomProtectionUtils.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils;
+
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.util.List;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.utils.CommonConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public final class OomProtectionUtils {
+  private static final Logger LOGGER = LoggerFactory.getLogger(OomProtectionUtils.class);
+
+  private OomProtectionUtils() {
+  }
+
+  public static boolean hasDisableAdaptiveIhopFlag(List<String> jvmInputArgs) {
+    if (jvmInputArgs == null || jvmInputArgs.isEmpty()) {
+      return false;
+    }
+    for (String arg : jvmInputArgs) {
+      if ("-XX:-G1UseAdaptiveIHOP".equals(arg)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * If the required GC flag is missing, disables OOM protection by flipping the config flag to false.
+   * @return true if the method disabled OOM protection; false otherwise
+   */
+  public static boolean enforceIhopGcOrDisableOom(PinotConfiguration config, List<String> jvmInputArgs) {
+    boolean hasFlag = hasDisableAdaptiveIhopFlag(jvmInputArgs);
+    if (!hasFlag) {
+      LOGGER.warn("Required GC option -XX:-G1UseAdaptiveIHOP not found. Disabling OOM protection.");
```

Comment:
Does this mean our current thread resource usage tracking and query OOM kill feature work only with G1GC and not with any others?